0099-Replacing bash as the default shell.md

```diff
@@ -0,0 +1,120 @@
+
+[reference](https://github.com/oilshell/oil/wiki/Migration-Guide)
+
+Some more background context. Originally this is coming from this [PR](https://github.com/NixOS/nixpkgs/pull/105233) by zimbatm. After seeing this, I thought I would implement the changes needed to make the switch and generate a discussion in a PR. Somebody brought to my attention that this change is significant enought that it needs to go through an RFC.
+
+## Examples and Interactions
+
+[examples-and-interactions]: #examples-and-interactions
+
+Some extracts from the post [why use oil](https://www.oilshell.org/why.html)
+
+- better error handling. One of the confusing things about bash is how to handle errors in scripts. `err_exit` was a try at improving that but it has many pitfalls. Oil aims to have more straightforward error handling. Any error will exit, if you do not want that behavior there is the `try my_function` form.
+
+`mkdir /tmp/dest && cp foo /tmp/dest` becomes simply
+
+```Shell
+mkdir /tmp/dest
+cp foo /tmp/dest
+```
+
+- Introduce functions that need to explicitely define their parameters
+
+```Shell
+proc f(first, @rest) { # @ means "the rest of the arguments"
+  write -- $first
+  write -- @rest # @ means "splice this array"
+}
+```
+
+instead of the traditional
+
+```Shell
+f() {
+  echo $1
+  shift 1
+  echo "$@"
+}
+```
+
+those functions are called procs and their variables don't mess with their outer scope.
+
+- Oil has proper associative arrays (dictionaries or hashmaps) that don't have the problems that bash's have.
+
+- No more quotes to prevent splitting
+
+## Drawbacks
+
+[drawbacks]: #drawbacks
+
+- Currently oilshell has a bus factor of 1. While Andrew has been very motivated and responsive thus far, this could change in the future.
+- Oil shell is pre 1.0 so the syntax might change.
+- Parts of oil are rewritten in c++, while this could be good for performance, new user might discover bugs that are really hard to debug.
+- Oil has had little usage so far, there might be bugs that will just be discovered by having a large user base.
+- Using osh, could be straightforward, but using oil might take more work as basically the existing bash code need to rewritten with `strict_errexit`.
+- The oil documentation is lacking at the moment.
+- Oil has really lofty goals, which I think it's good, but some potential solutions are not documented or not implemented. One example is eggex which are meant to be a replacement for reggexes.
+
+## Alternatives
+
+[alternatives]: #alternatives
+
+- I'm not aware of any other alternative shell that could enable a smooth transition with bash. [Here](https://github.com/oilshell/oil/wiki/Alternative-Shells) is a list of alternative shells.
+
+## Unresolved questions
```

Comment:
I would like to see some sort of performance estimate. I suspect it will be trivial but it would be nice to do a sanity check with some number of packages just to validate.

Reply:
Yes, I think some kind of performance eval is in order.  As noted in https://matthewbauer.us/blog/avoid-subshells.html, this can really add up in the stdenv.